finetune/input_pipeline.py

```diff
@@ -187,18 +193,21 @@ def _dataset_with_targets(self, Xs, Y, train, context=None, update_hook=None):
                     class_counts=class_counts
                 )
         shape_def = self.feed_shape_type_def()
-        return Dataset.from_generator(
+        dataset_gen = Dataset.from_generator(
             lambda: self.wrap_tqdm(dataset_encoded(), train, update_hook=update_hook), *shape_def
         )
+        return dataset_gen
 
     def _dataset_without_targets(self, Xs, train, context=None, update_hook=None):
         if context is not None:
             # we assume that X must have known length if we also provide context so this is safe
-            if callable(Xs):
+            if callable(Xs) and callable(context):
```

Comment:
Eventually (before merge) we'll also need this to handle the case where context is None

Reply:
it's handled in the other if statement